core/src/test/scala/org/locationtech/rasterframes/encoders/DevSpec.scala

```diff
@@ -0,0 +1,270 @@
+/*
+ * This software is licensed under the Apache 2 license, quoted below.
+ *
+ * Copyright 2019 Astraea, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     [http://www.apache.org/licenses/LICENSE-2.0]
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.locationtech.rasterframes.encoders
+
+import cats.data
+
+import geotrellis.proj4._
+import geotrellis.raster.{CellSize, CellType, Dimensions, TileLayout, UShortUserDefinedNoDataCellType}
+import geotrellis.layer._
+import geotrellis.vector.{Extent, ProjectedExtent}
+import org.apache.spark.sql.{Dataset, Encoder, Row}
+import org.apache.spark.sql.catalyst.encoders.{ExpressionEncoder, encoderFor}
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, BoundReference, Expression, IsNull, KnownNotNull}
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+import org.locationtech.rasterframes.{TestData, TestEnvironment}
+import org.locationtech.rasterframes.model.{CellContext, TileContext, TileDataContext}
+import org.scalatest.Assertion
+
+
+class DevSpec extends TestEnvironment {
+  import TestData._
+
+  describe("home rolled encoders") {
+    import spark.implicits._
+    it("matches spark serializers") {
+      val home = Person.encoder
+      val mySerializerDescription = home.treeString
+      info(mySerializerDescription )
+
+      val baked = implicitly[Encoder[Person]].asInstanceOf[ExpressionEncoder[Person]].objSerializer
+      val sparkSerializerDescription = baked.treeString
+      info(sparkSerializerDescription )
+
+      mySerializerDescription shouldBe sparkSerializerDescription
+    }
+
+    it ("Round-trip Person through DataFrame") {
+      implicit def en: Encoder[Person] = {
+        new ExpressionEncoder[Person](Person.encoder, Person.decoder, typeToClassTag[Person])
+      }
+      val ds = Seq(Person("Bob", 11), Person("Eugene", 10)).toDS
+
+      ds.printSchema()
+      ds.show()
+
+      val df = ds.toDF()
+
+      val me = df.as[Person].first()
+      info(me.toString)
+    }
```

Comment:
This is an attempt to mirror and break up the logic of construction ExpressionEncoder from spark `ScalaReflection` to see if we can build a type-class driven way to compose the encoders similar to CatalystSerializer. In this case the encoder for `Person` and `School` where built up "by hand" in this manner showing that it's at least possible.

The key motivator for this exercise is that the way spark constructs serializer/deserializer for case classes behaves well with regard to the problem we're seeing with ProjectedRasterTile. That is, it handles blowing out the members of Person struct to columns when its at row level and then recombining them into the java type for DataSet.